Observer/ImportRates.php

```diff
@@ -358,41 +345,84 @@ private function makeOperation($rates, $bulkUuid): OperationInterface {
         return $this->operationFactory->create($data);
     }
 
+    /**
+     * @param string $bulkUuid
+     * @param array $operations
+     * @param string $bulkDescription
+     * @throws LocalizedException
+     */
+    private function schedule($bulkUuid, $operations, $bulkDescription): void
+    {
+        $userId = $this->userContext->getUserId();
+        $result = $this->bulkManagement->scheduleBulk($bulkUuid, $operations, $bulkDescription, $userId);
+
+        if (! $result) {
+            throw new LocalizedException(
+                __('Something went wrong while processing the request.')
+            );
+        }
+    }
+
     /**
      * Purge existing rule calculations and rates
      *
      * @return void
+     * @throws LocalizedException
      */
-    private function _purgeRates()
+    private function purgeRates()
     {
         $rateModel = $this->rateFactory->create();
         $rates = $rateModel->getExistingRates();
 
-        foreach ($rates as $rate) {
-            $calculations = $rateModel->getCalculationsByRateId($rate);
-
-            try {
-                foreach ($calculations->getItems() as $calculation) {
-                    // @codingStandardsIgnoreStart
-                    $calculation->delete();
-                    // @codingStandardsIgnoreEnd
-                }
-            } catch (\Exception $e) {
-                $this->messageManager->addErrorMessage($e->getMessage());
+        if ($rule = $rateModel->getRule()) {
```

Comment:
It appears that this if statement always evaluates to true, even if there is not a TaxJar rate stored in the database. It may be unnecessary.